### PR TITLE
fix: popper content type check

### LINKED ISSUE #17030: [TypeScript] [popover] Popover 气泡卡片 effect Type bug
Bug Type: **`TypeScript`**

## Environment

- Vue Version: `3.4.21`
- Element Plus Version: `2.7.2`
- Browser / OS: `edge `
- Build Tool: `Vite`

### Related Component

- `el-popover`

### Reproduction Link

[Docs](https://element-plus.org/zh-CN/component/popover.html#attributes)

### Steps to reproduce

```vue
<template>
    <el-popover
         placement="left"
         effect="dark"
         :width="250">
    </el-popover>
</template>
```

经过TS约束检查时发现, effect 可接受的参数仅为 string
与文档中提到的,支持element theme 主题 dark 和 light 表现不一致;

### What is Expected?

期望 修复 el-popover 组件对 effect 属性的约束. 将错误的 约束类型 "string" 更改为  "dark" | "light"

### What is actually happening?

当前el-popover 组件的 effect属性只允许使用 "string" 作为属性值

### Additional comments

_(empty)_

### PR DESCRIPTION
**Pull request by JiuRanYa**
Please make sure these boxes are checked before submitting your PR, thank you!

fix #17030

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

**Comment by btea** [comment]
Hello, your modification seems to cause TS errors.

**Comment by JiuRanYa** [comment]
@btea Sorry to reply too late. I have fixed all the type related to the popper `effect`.

File: packages/components/popper/src/popper.ts [hunk 1]
```diff
@@ -10,6 +10,7 @@ export const Effect = {
   LIGHT: 'light',
   DARK: 'dark',
 } as const
+export type EffectType = typeof Effect[keyof typeof Effect]
```
Comment (tolking) [inline packages/components/popper/src/popper.ts:13]: `EffectType` as same as `PopperEffect`

Comment (JiuRanYa) [reply packages/components/popper/src/popper.ts:13]: Yeah, I have noticed that. So i just need to replace all `EffectType` with `PopperEffect`?

Comment (tolking) [reply packages/components/popper/src/popper.ts:13]: Yep.

**Review by tolking — APPROVED** [review]
LGTM.

**Review by btea — APPROVED** [review]
Thanks!

**Pull request merged**